pyiron_base/jobs/job/extension/server/generic.py

```diff
@@ -91,6 +102,7 @@ def __init__(
         self._memory_limit = None
         self._host = self._init_host(host=host)
         self._run_mode = Runmode()
+        self._executor = None
```

Comment:
```suggestion
        self._executor: Union[Executor, None] = None
        self._future: Union[Future, None] = None
```

Readability nit: slid `_future` up to be close to the thing it's related to and added type hint.